### DISCUSSION TITLE
Installing Mealie on Raspberry Pi

### DISCUSSION THREAD

**Discussion by palaplan** [mealie-recipes/mealie#2283] [Q&A]
I am trying to run a Docker image of Mealie on my Raspberry Pi 4 Model B. I copied Raspberry Pi OS on a Micro SD card using the [Raspberry Pi Imager](https://www.raspberrypi.com/software/) (the default version is the 32 bits OS). I then installed Docker following [these instructions](https://www.simplilearn.com/tutorials/docker-tutorial/raspberry-pi-docker) but was not able to run 
```
docker run \
    -p 9925:80 \
    -v `pwd`:'/app/data/' \
    hkotel/mealie:latest
```
the error I got was
```
docker: no matching manifest for linux/arm/v7 in the manifest list entries. 
```

I then found the Docker image of the [version 0.4.3](https://hub.docker.com/layers/hkotel/mealie/v0.4.3/images/sha256-ede8e91179327d6feaffd9503f668e6e9ba763762c3203c537e1e10aed299f7e?context=explore) that support `linux/arm/v7` and was able to run it running the following:
```
docker run \
    -p 9925:80 \
    -v `pwd`:'/app/data/' \
    hkotel/mealie:v0.4.3
```
So it runs nicely now but the version is now almost 2 years old.

I am assuming that the Docker images now only support 64 bit systems and that I will need to install the 64 bits version of the Raspberry Pi OS in order to use the latest version, is this correct?

**Comment by hay-kot** [comment]
> I am assuming that the Docker images now only support 64 bit systems and that I will need to install the 64 bits version of the Raspberry Pi OS in order to use the latest version, is this correct?

That's correct. arm/v7 builds were very unpredictable and would fail _all the time_ for seemingly no reason and would eat up developer time so we dropped support. Others have reported that using a 64bit OS on the Pi works great.

**Reply by lowlyocean** [reply]
For folks already using their RPI to serve other apps, upgrading to 64 bit OS is not ideal. I encountered the error mentioned [here](https://hay-kot.github.io/mealie/changelog/v0.5.0/#:~:text=Mealie%20will%20no%20longer%20build%20in%20CI/CD%20due%20to%20a%20issue%20with%20the%20rust%20compiler%20on%2032%20bit%20devices) when installing docker-compose on the RPI, but workaround was just to use --only-binary=cryptography . Could similar approach be used to fix the CI/CD so that we can restore availability of armv7 mealie images?

Edit: I suppose you may have tried already --only-binary flag , but since you use Poetry I am guessing due to [this issue](https://github.com/python-poetry/poetry/issues/6798) you don't have ability for this simple workaround that you would have if you used Pip instead
